codes/models/SRRaGAN_model.py

```diff
@@ -359,9 +376,14 @@ def optimize_parameters(self, gen, step):
                         l_g_total += l_g_lpips
                         self.log_dict['l_g_lpips'] += l_g_lpips.item()
                     # G gan + cls loss
-                    pred_g_fake = self.netD(self.fake_H)
-                    pred_d_real = self.netD(self.var_ref).detach()
-
+                    if self.use_frequency_separation:
+                        pred_g_fake = self.netD(self.filter_high(self.fake_H))
+                    else:
+                        pred_g_fake = self.netD(self.fake_H)
+                    if self.use_frequency_separation:
+                        pred_d_real = self.netD(self.filter_high(self.var_ref)).detach()
+                    else:
+                        pred_d_real = self.netD(self.var_ref).detach()
```

Comment:
Can these be combined into one if statement?

Reply:
Will take a look into it!

Reply:
Changes made